drivers/cmd/kubernetes/main.go

```diff
@@ -104,14 +130,16 @@ func getJobLog(m *dipper.Message) {
 
 		podlogs := map[string]string{}
 		for _, container := range append(pod.Spec.InitContainers, pod.Spec.Containers...) {
-			stream, err := client.GetLogs(pod.Name, &corev1.PodLogOptions{Container: container.Name}).Stream()
-			if err != nil {
-				podlogs[container.Name] = fmt.Sprintf("Error: unable to fetch the logs from the container %s.%s", pod.Name, container.Name)
-				messages = append(messages, podlogs[container.Name])
-				log.Warningf("[%s] unable to fetch the logs for the pod %s container %s: %+v", driver.Service, pod.Name, container.Name, err)
-				returnStatus = StatusFailure
-			} else {
-				func(stream io.ReadCloser) {
+			func() {
+				ctx, cancel := context.WithTimeout(context.Background(), k8stimeout*time.Second)
+				defer cancel()
+				stream, err := client.GetLogs(pod.Name, &corev1.PodLogOptions{Container: container.Name}).Stream(ctx)
+				if err != nil {
+					podlogs[container.Name] = fmt.Sprintf("Error: unable to fetch the logs from the container %s.%s", pod.Name, container.Name)
+					messages = append(messages, podlogs[container.Name])
+					log.Warningf("[%s] unable to fetch the logs for the pod %s container %s: %+v", driver.Service, pod.Name, container.Name, err)
+					returnStatus = StatusFailure
+				} else {
 					defer stream.Close()
```

Comment:
Can we move this to line 136 to have the stream close regardless if we get a stream back from logs or not?

Reply:
You are right. It should be called wether or not err is nil.